src/Compilers/CSharp/Portable/Lowering/MethodToClassRewriter.cs

```diff
@@ -124,6 +124,14 @@ private ImmutableArray<LocalSymbol> RewriteLocals(ImmutableArray<LocalSymbol> lo
             return newLocals.ToImmutableAndFree();
         }
 
+        public override BoundNode VisitLoweredSwitchExpressionArm(BoundLoweredSwitchExpressionArm node)
+        {
+            return node.Update(
+                RewriteLocals(node.Locals),
```

Comment:
Locals in `BoundScope` are treated differently in each subclass, so this might be wrong. The bootstrap failure is due to this assertion:

https://github.com/dotnet/roslyn/blob/724dfa4863ec244ca3d40d7ded4d0fee8b94e7bc/src/Compilers/CSharp/Portable/Lowering/IteratorRewriter/IteratorMethodToStateMachineRewriter.cs#L361

Fired on:

https://github.com/dotnet/roslyn/blob/724dfa4863ec244ca3d40d7ded4d0fee8b94e7bc/src/Features/CSharp/Portable/DocumentationComments/CSharpDocumentationCommentSnippetService.cs#L149